client/src/main/java/io/pravega/client/segment/impl/ConditionalOutputStreamImpl.java

```diff
@@ -66,7 +66,7 @@ public boolean write(ByteBuffer data, long expectedOffset) throws SegmentSealedE
                     .throwingOn(SegmentSealedException.class)
                     .run(() -> {
                         if (client == null || client.isClosed()) {
-                            client = new RawClient(controller, connectionFactory, segmentId);
+                            client = new RawClient(controller.getEndpointForSegment(segmentId.getScopedName()).get(), connectionFactory, segmentId);
```

Comment:
controller.getEndpointForSegment is a remote call and you are blocking on it here before passing it as constructor param. You can make it asynchronous by doing 
controller.getEndpointForSegment().thenApply(uri -> new RawClient(uri, connectionFactory, segmentId))

Reply:
thanks Shivesh,for the comments do you mean to replace with the following statement:
          controller.getEndpointForSegment(**segmentId.getScopedName()**).thenApply(uri -> new RawClient(uri, connectionFactory, segmentId));